test/libsolidity/smtCheckerTests/inheritance/diamond_super_1.sol

```diff
@@ -0,0 +1,31 @@
+pragma experimental SMTChecker;
+contract A {
+	function f() public virtual returns (uint256 r) {
+		return 1;
+	}
+}
+
+
+contract B is A {
+	function f() public virtual override returns (uint256 r) {
+		return super.f() + 2;
+	}
+}
+
+
+contract C is A {
+	function f() public virtual override returns (uint256 r) {
+		return super.f() + 4;
+	}
+}
+
+
+contract D is B, C {
+	function f() public override(B, C) returns (uint256 r) {
+		r = super.f() + 8;
+		assert(r == 15); // should hold
+		assert(r == 13); // should fail
+	}
+}
```

Comment:
`super.f() + constant` doesn't really test the order of traversal, i.e., both `D.f -> B.f -> C.f -> A.f` and `D.f -> C.f -> B.f -> A.f` will give the same result.

Can you make `f` return `1`, `super.f() + 2`, `2 * (super.f() + 4)` and `super.f() + 8` respectively?

Reply:
Right, I was rather testing that nothing gets skipped, but I can also add that

Reply:
Added